5_data_munge/src/munge_wqp_files.R

```diff
@@ -0,0 +1,102 @@
+# mungers for WQP data
+
+filter_site_types <- function(in_ind, sites_ind, keep_types, out_ind) {
+  
+  sites <- feather::read_feather(sc_retrieve(sites_ind, remake_file = 'getters.yml')) %>%
+    filter(ResolvedMonitoringLocationTypeName %in% keep_types) %>%
+    pull(MonitoringLocationIdentifier)
+  
+  dat <- readRDS(sc_retrieve(in_ind, remake_file = 'getters.yml')) %>%
+    filter(MonitoringLocationIdentifier %in% sites)
+  
+  saveRDS(dat, as_data_file(out_ind))
+  gd_put(out_ind)
+  
+  
+}
+
+munge_wqp_withdepths <- function(in_ind, min_value, max_value, max_daily_range, out_ind) {
+  dat <- readRDS(sc_retrieve(in_ind, remake_file = 'getters.yml')) %>%
+    filter(!is.na(`ResultMeasure/MeasureUnitCode`)) %>%
+    filter(ActivityMediaName %in% 'Water') %>%
+    filter(!is.na(`ActivityDepthHeightMeasure/MeasureValue`)|
+             !is.na(`ResultDepthHeightMeasure/MeasureValue`)|
+             !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`)|
+             !is.na(`ActivityBottomDepthHeightMeasure/MeasureValue`))
+  # create depth column
+  dat <- mutate(dat, sample_depth = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureValue`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureValue`
+  ))
+  
+  dat <- mutate(dat, sample_depth_unit_code = case_when(
+    !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureUnitCode`,
+    is.na(`ActivityDepthHeightMeasure/MeasureValue`) & is.na(`ResultDepthHeightMeasure/MeasureValue`) & is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityBottomDepthHeightMeasure/MeasureUnitCode`
```

Comment:
This guidance in `?case_when` makes me think you could get away with a simpler code block above.
```r
# Like an if statement, the arguments are evaluated in order, so you must
# proceed from the most specific to the most general. This won't work:
case_when(
  TRUE ~ as.character(x),
  x %%  5 == 0 ~ "fizz",
  x %%  7 == 0 ~ "buzz",
  x %% 35 == 0 ~ "fizz buzz"
)
```

So I'm fairly confident you could do
```r

dat <- mutate(dat, sample_depth_unit_code = case_when(
     !is.na(`ActivityDepthHeightMeasure/MeasureValue`) ~ `ActivityDepthHeightMeasure/MeasureUnitCode`,
     !is.na(`ResultDepthHeightMeasure/MeasureValue`) ~ `ResultDepthHeightMeasure/MeasureUnitCode`,
     !is.na(`ActivityTopDepthHeightMeasure/MeasureValue`) ~ `ActivityTopDepthHeightMeasure/MeasureUnitCode`,
     TRUE ~ `ActivityBottomDepthHeightMeasure/MeasureUnitCode`
))
```

Reply:
double checking with a more similar example:
```r
> tibble(a=c(1, NA, NA, NA, 1), b=c(NA, 2, NA, 2, NA), c=c(NA, NA, 3, 3, 3)) %>%
+     mutate(best = case_when(!is.na(a) ~ a, !is.na(b) ~ b, !is.na(c) ~ c, TRUE ~ 0))
# A tibble: 5 x 4
      a     b     c  best
  <dbl> <dbl> <dbl> <dbl>
1     1    NA    NA     1
2    NA     2    NA     2
3    NA    NA     3     3
4    NA     2     3     2
5     1    NA     3     1
```